app/controllers/api/vms_controller.rb

```diff
@@ -351,7 +351,7 @@ def vm_event(vm, event_type, event_message, event_time)
     def retire_vm(vm, id, data)
       desc = "#{vm_ident(vm)} retiring"
       desc << " on #{data['date']}" if Hash(data)['date'].present?
-      retire_resource(:vms, id, data)
+      method(:retire_resource).super_method.call(:vms, id, data)
```

Comment:
Not crazy about starting to inline that code (there goes the nice cookie cutter).

Another option is to write a wrapper so simplify the calling sequence, i.e. a method: _call_super(:method, *args)_
which can do what you have above or BaseController.instance_method(method).bind(self).call(*args). though preferring your approach (with _super_method_), but updated with more generalized arguments.   

Thanks.